aya-log-common/src/lib.rs

```diff
@@ -72,10 +72,28 @@ mod userspace {
 
     unsafe impl aya::Pod for RecordField {}
     unsafe impl aya::Pod for ArgType {}
+    unsafe impl aya::Pod for DisplayHint {}
+}
+
+/// All display hints
+#[repr(usize)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum DisplayHint {
+    /// Default string representation.
+    Default = 1,
+    /// `:x`
+    LowerHex,
+    /// `:X`
+    UpperHex,
+    /// `:ipv4`, `:IPv4`
+    IPv4,
+    /// `:ipv6`, `:IPv6`
+    IPv6,
 }
 
-struct TagLenValue<'a, T> {
+pub struct TagLenValue<'a, T> {
     tag: T,
+    hint: DisplayHint,
```

Comment:
This doesn't look good - you turned it into `TagHintLenValue` 😅

I instead think the layout should be this:

```
[record header]
[arguments_and_hints]
```

where each `arguments_and_hints` is a sequence of `TagLenValue` where tags can be either `RecordField::Argument` or `RecordField::DisplayHint` (optional).

Whenever you find a `RecordField::Argument` if it was preceded by a `RecordField::DisplayHint` you apply the hint, else you apply the default hint. 

And for each argument, instead of writing `ArgType` as a tag len value (24 bytes), you can simply just start each argument with its arg type (8 bytes), so we don't waste space.

Reply:
> This doesn't look good - you turned it into `TagHintLenValue` 😅
> 
> Instead I think the layout should be this:
> 
> ```
> [record header]
> [arguments_and_hints]
> ```
> 
> where each `arguments_and_hints` is a sequence of `TagLenValue` where tags can be either `RecordField::Argument` or `RecordField::DisplayHint` (optional).
> 
> Whenever you find a `RecordField::Argument` if it was preceded by a `RecordField::DisplayHint` you apply the hint, else you apply the default hint.

Actually `[arguments_and_hints]` should probably use another enum, eg: 

```
enum Argument {
   DisplayHint,
   Value
}
```

and you loop until you've read `num_args` (as specified in `[record header]`) applying hints as you find them